examples/stable-diffusion/train_text_to_image_sdxl.py

```diff
@@ -666,7 +672,10 @@ def main(args):
 
     # If passed along, set the training seed now.
     if args.seed is not None:
-        set_seed(args.seed)
+        if args.full_determinism:
+            enable_full_determinism(args.seed)
+        else:
+            set_seed(args.seed)
```

Comment:
should we set_seed as long as see is not none, and if full determnistic, then set enable_full_determinism ?
also i am not sure if enable_full_determinism is supported by us

Reply:
this code is part of the transformer trainer.py, so we were just using this whole thing for work for HPU. If we need to add specific code for stable-diffusion, I think it's best to what's needed for HPU.  

if args.deterministic: 
 torch.use_deterministic_algorithms(True) and set_seed()